### ISSUE TITLE
SWR and @here/maps-api-for-javascript prod build

### ISSUE THREAD

**Issue by ivanzorya** [vercel/next.js#66182]
### Link to the code that reproduces this issue

https://github.com/ivanzorya/here-bungle-comp-issue-example

### To Reproduce

1. npm config set @here:registry https://repo.platform.here.com/artifactory/api/npm/maps-api-for-javascript/
2. Build prod app - next build
3. Run prod app - next start
4. Receive error for map bundle
5. Uncomment swcMinify: false in next.config.js, build and run again - works fine.

### Current vs. Expected behavior

SWR not correctly prepare prod build for HERE lib

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #35~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue May  7 09:00:52 UTC 2
  Available memory (MB): 27877
  Available CPU cores: 12
Binaries:
  Node: 20.12.1
  npm: 10.5.0
  Yarn: 1.22.22
  pnpm: N/A
Relevant Packages:
  next: 14.2.3 // Latest available version is detected (14.2.3).
  eslint-config-next: N/A
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.1.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

SWC

### Which stage(s) are affected? (Select all that apply)

next build (local), next start (local)

**Comment by kdy1** [comment]
It seems like such npm package does not exist. https://www.npmjs.com/package/@here/maps-api-for-javascript

**Comment by ivanzorya** [comment]
@kdy1 

npm config set @here:registry https://repo.platform.here.com/artifactory/api/npm/maps-api-for-javascript/

**Comment by kdy1** [comment]
Can you reproduce it on canary? It seems like the check in the issue template is gone

**Comment by ivanzorya** [comment]
Still reproducible on canary version

**Comment by kdy1** [comment]
Can you update the repro, please?

**Comment by ivanzorya** [comment]
```
Operating System:
  Platform: linux
  Arch: x64
  Version: #35~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue May  7 09:00:52 UTC 2
  Available memory (MB): 27877
  Available CPU cores: 12
Binaries:
  Node: 20.12.1
  npm: 10.5.0
  Yarn: 1.22.22
  pnpm: N/A
Relevant Packages:
  next: 15.0.0-canary.5 // Latest available version is detected (15.0.0-canary.5).
  eslint-config-next: N/A
  react: 19.0.0-rc-f994737d14-20240522
  react-dom: 19.0.0-rc-f994737d14-20240522
  typescript: 5.1.3
Next.js Config:
  output: N/A
```
Done

**Comment by kdy1** [comment]
It was fixed on `v15.0.0-canary.12`, which includes latest minifier bugfixes. The `swc_core` update had a regression and it was reverted, but I fixed the regression so [the swc_core update PR](https://github.com/vercel/turbo/pull/8395) will fix this issue.

**Comment by ivanzorya** [comment]
Works fine with `v15.0.0-canary.12`. Thanks.
